### PR TITLE
Fix C++14 compilation errors and upgrades EAPI to 6

### PR DESCRIPTION
**Pull request by kolodziej**
Assign-to: @SoapGentoo

### CODE REVIEW

File: dev-db/xbase/xbase-3.1.2-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+PATCHES=(
+		"${FILESDIR}/${P}-fixconfig.patch"
+		"${FILESDIR}/${P}-gcc44.patch"
+		"${FILESDIR}/${PN}-2.0.0-ppc.patch"
+		"${FILESDIR}/${P}-xbnode.patch"
+		"${FILESDIR}/${P}-lesserg.patch"
+		"${FILESDIR}/${P}-outofsource.patch"
+		"${FILESDIR}/${P}-gcc47.patch"
+		"${FILESDIR}/${P}-gcc-version.patch"
+		"${FILESDIR}/${P}-cpp14.patch" # bug #594174
```
Comment (SoapGentoo) [inline dev-db/xbase/xbase-3.1.2-r1.ebuild]: too much indentation (one tab)

File: dev-db/xbase/xbase-3.1.2-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	emake DESTDIR="${D}" install
+	# media-tv/linuxtv-dvb-apps collision, bug #208596
+	mv "${ED}/usr/bin/zap" "${ED}/usr/bin/${PN}-zap" || die
+
+	if use doc; then
+		einstalldocs
+		insinto /usr/share/doc/${PF}/examples
+		doins examples/*
+	fi
+}
```
Comment (SoapGentoo) [inline dev-db/xbase/xbase-3.1.2-r1.ebuild]: change this phase to
```
src_install() {
	if use doc; then
		HTML_DOCS+=( html/. )
		dodoc -r examples
		docompress -x /usr/share/doc/${PF}/examples
	fi
	default

	# media-tv/linuxtv-dvb-apps collision, bug #208596
	mv "${ED%/}"/usr/bin/{,${PN}-}zap || die
}
```
then nix the initial `HTML_DOCS` on line 30, the reason being that in general `einstalldocs` should be called unconditionally

File: dev-libs/tvision/tvision-2.1.0_pre2-r4.ebuild [hunk 3]
```diff
@@ -0,0 +1,47 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit eutils multilib
```
Comment (SoapGentoo) [inline dev-libs/tvision/tvision-2.1.0_pre2-r4.ebuild]: don't need either of these inherits

File: dev-libs/tvision/tvision-2.1.0_pre2-r4.ebuild [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+src_configure() {
+	./configure --fhs || die
+}
+
+src_install() {
+	emake DESTDIR="${D}" install \
+		prefix="\${DESTDIR}/usr" \
+		libdir="\$(prefix)/$(get_libdir)" \
+		|| die
```
Comment (SoapGentoo) [inline dev-libs/tvision/tvision-2.1.0_pre2-r4.ebuild]: don't need `|| die`, `emake` dies on its own

File: dev-db/xbase/xbase-3.1.2-r1.ebuild [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+	"${FILESDIR}/${PN}-2.0.0-ppc.patch"
+	"${FILESDIR}/${P}-xbnode.patch"
+	"${FILESDIR}/${P}-lesserg.patch"
+	"${FILESDIR}/${P}-outofsource.patch"
+	"${FILESDIR}/${P}-gcc47.patch"
+	"${FILESDIR}/${P}-gcc-version.patch"
+	"${FILESDIR}/${P}-cpp14.patch" # bug #594174
+)
+
+HTML_DOCS=( html/.)
```
Comment (a17r) [inline dev-db/xbase/xbase-3.1.2-r1.ebuild]: This line is redundant when I look at src_install() section.

File: dev-db/xbase/xbase-3.1.2-r1.ebuild [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+	if use doc; then
+		HTML_DOCS+=( html/. )
+		dodoc -r examples
+		docompress -x /usr/share/doc/${PF}/examples
+	fi
+
+	default
+
+	# media-tv/linuxtv-dvb-apps collision, bug #208596
+	mv "${ED}/usr/bin/zap" "${ED}/usr/bin/${PN}-zap" || die
```
Comment (SoapGentoo) [inline dev-db/xbase/xbase-3.1.2-r1.ebuild]: `mv "${ED%/}"/usr/bin/{,${PN}-}zap || die`

**Comment by a17r** [comment]
@SoapGentoo: lgtm, what do you think?

File: dev-db/xbase/xbase-3.1.2-r1.ebuild [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+EAPI=6
+inherit autotools
+
+DESCRIPTION="xbase (i.e. dBase, FoxPro, etc.) compatible C++ class library"
+HOMEPAGE="http://linux.techass.com/projects/xdb/"
+SRC_URI="mirror://sourceforge/xdb/${PN}64-${PV}.tar.gz"
+
+SLOT="0"
+LICENSE="GPL-2"
```
Comment (SoapGentoo) [inline dev-db/xbase/xbase-3.1.2-r1.ebuild]: swap these two: first `LICENSE`, then `SLOT`

**Pull request closed without merging**